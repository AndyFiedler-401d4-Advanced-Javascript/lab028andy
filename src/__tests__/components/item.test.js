import React from 'react';
import renderer from 'react-test-renderer';
import Item from '../../components/item';

describe('<Item />', () => {
  it('can render initially in li with span, button', () => {
    let app = shallow(<Item item={({id: 0, text: 'Test', complete: false})} delete={()=>{}} toggleComplete={()=>{}} />);
    expect(app.find('li span').exists()).toBe(true);
    expect(app.find('li button').exists()).toBe(true);
  });

  it('can change completion status by click on span', () => {
    let toggled = false;
    let toggleComplete = () => {
      toggled = true;
    };
    let app = shallow(<Item item={({id: 0, text: 'Test', complete: false})} delete={()=>{}} toggleComplete={toggleComplete} />);
    let span = app.find('li span');
    span.simulate('click', {preventDefault: () =>{}});
    expect(toggled).toBeTruthy();
  });

  it('can change "delete" item by click button', () => {
    let deleted = false;
    let removeTodo = () => {
      deleted = true;
    };
    let app = shallow(<Item item={({id: 0, text: 'Test', complete: false})} removeTodo={removeTodo} toggleComplete={()=>{}} />);
    let button = app.find('li button.delete');
    button.simulate('click', {preventDefault: () =>{}});
    expect(deleted).toBeTruthy();
  });

  it('renders consistently', () => {
    const tree = renderer.create(<Item item={({id: 0, text: 'Test', complete: false})} delete={()=>{}} toggleComplete={()=>{}}/>).toJSON();
    // console.log(tree);
    expect(tree).toMatchSnapshot();
  });
});
