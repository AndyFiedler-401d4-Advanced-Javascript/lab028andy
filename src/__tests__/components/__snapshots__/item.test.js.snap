// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders consistently 1`] = `
<li
  className="complete-false"
>
  <span
    id={0}
    onClick={[Function]}
  >
    Test
  </span>
  <button
    className="delete"
    onClick={[Function]}
  >
    delete
  </button>
  <button
    className="details"
    onClick={[Function]}
  >
    details
  </button>
</li>
`;
